src/Parallel/Algorithm.hpp

```diff
@@ -774,36 +734,34 @@ AlgorithmImpl<ParallelComponent, tmpl::list<PhaseDepActionListsPack...>>::
     using databox_phase_type = tmpl::at_c<databox_phase_types, phase_index>;
     using databox_types_this_phase = typename databox_phase_type::databox_types;
 
-    using box_possibilities = std::conditional_t<
+    using potential_databox_indices = std::conditional_t<
         iter == 0_st,
-        tmpl::list<
-            std::integral_constant<size_t, 0_st>,
-            std::integral_constant<
-                size_t, tmpl::size<databox_types_this_phase>::value - 1_st>>,
-        tmpl::list<std::integral_constant<size_t, iter>>>;
+        tmpl::integral_list<size_t, 0_st,
+                            tmpl::size<databox_types_this_phase>::value - 1_st>,
+        tmpl::integral_list<size_t, iter>>;
     bool box_found = false;
-    tmpl::for_each<box_possibilities>([this, &box_found, &take_next_action,
-                                       &check_if_ready](
-                                          auto box_possibility_v) noexcept {
-      const size_t box_possibility = decltype(box_possibility_v)::type::value;
-      using this_databox =
-          tmpl::at_c<databox_types_this_phase, box_possibility>;
-      if (not box_found and
-          box_.which() ==
-              static_cast<int>(
-                  tmpl::index_of<variant_boxes, this_databox>::value)) {
-        box_found = true;
-        auto& box = boost::get<this_databox>(box_);
-        using local_this_action = tmpl::at_c<actions_list, iter>;
-        if (not check_if_ready(local_this_action{}, box)) {
-          take_next_action = false;
-          return;
-        }
-        performing_action_ = true;
-        ++algorithm_step_;
-        invoke_iterable_action<this_action, actions_list>(box);
-      }
-    });
+    tmpl::for_each<potential_databox_indices>(
+        [this, &box_found, &take_next_action,
+         &check_if_ready](auto box_possibility_v) noexcept {
+          const size_t box_possibility =
```

Comment:
I think this should be `constexpr` instead of `const` since you use this in a template parameter below. I'm not certain what the rules are and if the compilers follow them :package: 

[optional] Maybe `databox_index` or something like that to indicate in the name it's not a `DataBox` object or type?

Reply:
I'd like the [optional] to be done here and in the next commit (also for `box_possibility_v` -> `databox_index_v` or `potential_databox_index_v`